### DISCUSSION TITLE
Is Bidirectional with infinite scroll supported?

### DISCUSSION THREAD

**Discussion by affemaen** [TanStack/virtual#704] [Q&A]
Has anyone been able to create a bidirectional list with infinite scroll? From what I have found the scroll position is not restored when new items are added to the top. It would be really nice with an example in the docs. It's a pretty common use case for chats and cursor based navigation.

**Comment by piecyk** [comment]
@affemaen checkout the discussion at https://github.com/TanStack/virtual/discussions/477

**Reply by affemaen** [reply]
@piecyk thank you! Unfortunately the discussion seems a bit dead, strange since it's a pretty common use case. I'll see if I can get some time to dig deeper into it.

**Reply by piecyk** [reply]
Yep, for sure we need to add proper example, if something created quick WIP, hope this helps https://stackblitz.com/edit/tanstack-query-xrw3fp?file=src%2Fpages%2Findex.js

**Reply by affemaen** [reply]
Thx, looks great. I've done something similar trying to calculate the offset myself, wish it could be built in to the lib. Will look into it more and port it to Vue.

**Comment by issimin** [comment]
If there is a big deviation between estimateSize and the actual size, it will not scroll to the previous position after pulling up, may I ask if anyone can solve this problem? Thx.

**Reply by piecyk** [reply]
hmm most cases it should work correctly, maybe could be problematic when element has more size than scroll rect, can you provide an example?

**Reply by issimin** [reply]
Thanks for your reply. After adjusting itemsize to 250 in this case, there will be some deviation after scrolling up and jumping。https://stackblitz.com/edit/tanstack-query-xrw3fp?file=src%2Fpages%2Findex.js

**Comment by henryogunrinde** [comment]
If anyone still wants this, check this chat implementation here: https://github.com/TanStack/virtual/discussions/195#discussioncomment-11228646
